dev-tools/docker/README.md

```diff
@@ -0,0 +1,137 @@
+# Summary of Docker images in the ml-dev namespace
+
+## Repository: ml-linux-dependency-build
+
+### Latest version: pytorch_viable_strict
+
+### Comments
+A Docker image that can be used to compile the machine learning
+C++ code for Linux with the latest build of the PyTorch release branch.
+
+This image is intended to be built regularly in order to flush out any
+issues we may have with the latest PyTorch release as early as possible.
+To avoid swamping the Docker registry with numerous images we re-use the
+same tag - pytorch_viable_strict - for the image.
+
+### Depends on: ml-linux-dependency-builder:1
+
+### Build script: dev-tools/docker/build_pytorch_linux_build_image.sh
+
+
+## Repository: ml-linux-dependency-builder
+
+### Latest version: 1
+
+### Comments
+A Docker image that can be used to compile the dependencies of
+the machine learning C++ code for Linux, e.g. PyTorch or Boost, in order
+to test the latest versions of them.
+
+This image is not intended to be built regularly. It is based on the latest
+`ml-linux-build` image, but with extra dependencies added in order to build
+PyTorch.  Therefore, When changing any of the tools or 3rd party components
+required to build the machine learning C++ code dependencies increment the
+version, change the Dockerfile and build a new image to be used for subsequent
+builds on this branch. Then update the version to be used for builds in docker
+files that refer to it.
```

Comment:
nit: it would be easier to parse if you would format it as a numbered list of things to do.